OMCompiler/Compiler/BackEnd/BackendDAE.mo

```diff
@@ -815,7 +815,7 @@ type SparseColoring = list<list< .DAE.ComponentRef>>;   // colouring
 /*
   Type only for transformation from analytical to structural singularity.
 */
-type LinearIntegerJacobianRow = array<Integer>;                       // Actual jacobian entries
+type LinearIntegerJacobianRow = list<tuple<Integer, Integer>>;        // Actual jacobian entries sparse, <column, value>
```

Comment:
Might be better here to use an array of arrays, where the innermost array has two entries

Reply:
It has to be of dynamic length because i apply a gaussian algorithm on this. Every time i have to update a row i have to traverse every entry anyway and replace the value with the new one, the order does not matter since the actual column index is tracked inside. If i had an array i had to delete entries for which the value resolves to be zero and change the size over and over again.

See `SymbolicJacobian.solveLinearIntegerJacobianRowSingleStep` for the update of one single row.

Reply:
Ah if that is the case consider using DoubleEnded.List (That is the mutable list implementation). If the list datatype is immutable and is more adapted to functional programming